CLOUDP-260777: Plugin documentation (#9)
mongodb/atlas-cli-plugin-example: README.md

# Atlas CLI Plugin Example
This repository was created for developers who want to extend the [Atlas CLI](https://github.com/mongodb/mongodb-atlas-cli) with their own functionality. 

## How does it work?

> A plugin for the Atlas CLI is essentially a standalone CLI. When a plugin is installed, the Atlas CLI downloads a release asset from the plugin's GitHub repository and unpacks it into a plugin directory within the Atlas CLI's config folder. The plugin is then integrated as a standard subcommand, seamlessly extending the core functionality of the CLI. When a plugin command is invoked, the Atlas CLI executes the plugin's binary and forwards:

Reviewer: I find this a little hard to follow, what about something among the lines of:

```markdown
### Plugin Execution
A plugin for the Atlas CLI consists of a manifest file and a binary.
The manifest defines which subcommands the plugin exposes, and the binary is called when one of these subcommands is called.

When a subcommand is invoked, the Atlas CLI executes the plugin's binary and forwards:
- all arguments 
- environment variables
- standard input/output/error

### Plugin Installation
1. When a plugin is installed, the Atlas CLI takes the following steps:
download the release asset from the current OS from the plugin GitHub repository
2. unpack the assets (manifest+binary+optinal extras) into the Atlas CLI plugin directory
```

# Atlas CLI Plugin Example
This repository was created for developers who want to extend the [Atlas CLI](https://github.com/mongodb/mongodb-atlas-cli) with their own functionality. 

## How does it work?

> A plugin for the Atlas CLI is essentially a standalone CLI. When a plugin is installed, the Atlas CLI downloads a release asset from the plugin's GitHub repository and unpacks it into a plugin directory within the Atlas CLI's config folder. The plugin is then integrated as a standard subcommand, seamlessly extending the core functionality of the CLI. When a plugin command is invoked, the Atlas CLI executes the plugin's binary and forwards 

Reviewer 2: ```suggestion
A plugin for the Atlas CLI is essentially a standalone CLI. When a plugin is installed, the Atlas CLI downloads a release asset from the plugin's GitHub repository and unpacks it into a plugin directory within the Atlas CLI's config folder. The plugin is then integrated as a standard subcommand, seamlessly extending the core functionality of the CLI. When a plugin command is invoked, the Atlas CLI executes the plugin's binary and forwards:
```

- all arguments 
- environment variables
- standard input/output/error

> That allows the plugin to operate as if it were a native part of the Atlas CLI.

Reviewer 2: Suggest cutting this line.

Author: Done

## Developing a Plugin

> Plugins can be developed with any programming language, however, in this example plugin we are using Go. A plugin directory can contain any number of files, but it strictly requires two: an executable binary file and a `manifest.yml` file.

Reviewer 2: ```suggestion
Plugins can be developed with any programming language. A plugin directory can contain any number of files, but it strictly requires two: an executable binary file and a `manifest.yml` file.
```

commands: 
    example: 
        description: Root command of the atlas cli plugin example
```

| Field | Description | Optional |
| - | - | - |
| `name` | The name of the plugin | No
| `description` | A brief description of the plugin. | No
| `version` | The [semantic versioning](https://semver.org/) of the plugin.  | No
| `github.owner` | The GitHub repository owner | Yes
| `github.name`| The GitHub repository name | Yes
| `binary`| The name of the plugin's binary file | No
| `commands.<command_name>` | An object where the key is the command name and the value is a description of the command | No (a plugin needs at least one command)
| `commands.<command_name>.description` | A description of the command | No

> **Important:** 
> - The `version` field in the manifest file is crucial for the Atlas CLI to identify the installed plugin version and manage updates. Ensure that this field is always aligned with the plugin's release version to enable correct updates.
> - If your plugin is not part of a Github repository you can omit the entire `github` section. However, if your plugin is hosted on GitHub and you want to enable update functionality, you must include this section.

> You may notice that there is no `manifest.yml` file in this repository, only a `manifest.template.yml`. This is because we use a [GitHub Action](https://github.com/mongodb/atlas-cli-plugin-example/blob/master/.github/workflows/release.yml) to automatically generate the `manifest.yml` file. Every time a new release is published, the template is used to create `manifest.yml` file and the release's version is automatically inserted into the `version` field. The [GoReleaser](https://goreleaser.com/) is used to create release assets that contain the plugin's binary and the `manifest.yml`.

Reviewer 2: ```suggestion
You may notice that there is no `manifest.yml` file in this repository, only a `manifest.template.yml`. This is because we use a [GitHub Action](https://github.com/mongodb/atlas-cli-plugin-example/blob/master/.github/workflows/release.yml) to automatically generate the `manifest.yml` file. Every time a new release is published, the template is used to create the required `manifest.yml` file, and the release's version is automatically inserted into the `version` field. The [GoReleaser](https://goreleaser.com/) is used to create release assets that contain the plugin's binary and the `manifest.yml`.
```

# Atlas CLI Plugin Example
This repository was created for developers who want to extend the [Atlas CLI](https://github.com/mongodb/mongodb-atlas-cli) with their own functionality. 

## How does it work?

### Plugin Execution
A plugin for the Atlas CLI consists of a manifest file and a binary.

> The manifest defines which subcommands the plugin exposes, and the binary is called when one of these subcommands is called.

Reviewer 2: ```suggestion
The manifest defines which subcommands the plugin exposes, and the binary is called when one of these subcommands are called.
```

When a subcommand is invoked, the Atlas CLI executes the plugin's binary and forwards:
- all arguments 
- environment variables
- standard input/output/error

### Plugin Installation

> When a plugin is installed, the Atlas CLI takes the following steps:

Reviewer 2: ```suggestion
When you install a plugin, the Atlas CLI takes the following steps:
```

> 1. download the release asset for the users OS from the plugin GitHub repository

Reviewer 2: ```suggestion
1. download the release asset for the user's OS from the plugin GitHub repository
```
